sys/hashes/md5.c

```diff
@@ -232,15 +232,17 @@ void md5_init(md5_ctx_t *ctx)
     ctx->abcd[3] = 0x10325476;
 }
 
-void md5_update(md5_ctx_t *ctx, const uint8_t *data, size_t len)
+void md5_update(md5_ctx_t *ctx, const void *data, size_t len)
 {
+    const uint8_t *d = data;
+
     /* Add the new block's length to the total length. */
     ctx->len += (uint32_t)len;
 
     /* Copy the new block's data into the context block.
      * Call the permute() function whenever the context block is full. */
     for (size_t i = 0; i < len; i++) {
-        ctx->block[ctx->b_used] = data[i];
+        ctx->block[ctx->b_used] = d[i];
```

Comment:
I wonder if `ctx->block[ctx->b_used] = ((uint8_t*) data)[i];` wouldn't be the "better" solution, since one can see at one glance where `data` comes from.